graphql/server.go

```diff
@@ -165,14 +167,34 @@ func (c *conn) handleSubscribe(id string, subscribe *subscribeMessage) error {
 		ctx = batch.WithBatching(ctx)
 
 		start := time.Now()
-		span, ctx := opentracing.StartSpanFromContext(ctx, "thunder.subscription")
+
 		c.logger.StartExecution(ctx, tags, initial)
-		current, err := e.Execute(ctx, c.schema.Query, nil, query)
+
+		output := &ComputationOutput{
+			Metadata: make(map[string]interface{}),
+		}
+
+		var middlewares []MiddlewareFunc
+		middlewares = append(middlewares, c.middlewares...)
+		middlewares = append(middlewares, func(input *ComputationInput, output *ComputationOutput, next NextFunc) {
+			output.Current, output.Error = e.Execute(input.Ctx, c.schema.Query, nil, input.ParsedQuery)
+			next(input, output)
+		})
+
+		runMiddlewares(middlewares, &ComputationInput{
+			Ctx:         ctx,
+			Id:          id,
+			ParsedQuery: query,
+			Previous:    previous,
+			Query:       subscribe.Query,
```

Comment:
I think this is not the case because we don't have a printer method and we discard the underlying document node here: https://github.com/samsarahq/thunder/blob/master/graphql/parser.go#L344